### PR TITLE
Configurable hook script for recording synchronization

### LINKED ISSUE #1484: Configurable hook script for recording synchronization
**Is your feature request related to a problem? Please describe.**
In some hosting situations, mounting a shared storage or pushing recordings to PILOS via `rsync` might not be feasible.

**Describe the solution you'd like**
Operators should be able to write a custom hook script which pulls recordings to the spool folder. This script must be called before attempting to import recordings.
```[tasklist]
### Tasks
- [ ] Add `->withoutOverlapping()` to [https://github.com/THM-Health/PILOS/blob/8008f126af95c9e0c8d177382fbc6a6299006e69/app/Console/Kernel.php#L53
```

### PR DESCRIPTION
**Pull request by pizkaz**
# Fixes #1484 

## Type (Highlight the corresponding type)

-   Bugfix
-   **Feature**
-   Documentation
-   Refactoring (e.g. Style updates, Test implementation, etc.)
-   Other (please describe):

## Checklist

-   [x] Code updated to current develop branch head
-   [x] Passes CI checks
-   [x] Is a part of an issue
-   [x] Tests added for the bugfix or newly implemented feature, describe below why if not
-   [x] Changelog is updated
-   [x] Documentation of code and features exists

## Changes

- Introduces a new environment variable `RECORDING_IMPORT_BEFORE_HOOK` which can be set to a path to a script. If set, said script is called before importing new recordings from the spool directory. This can be handy in situations where one cannot configure BBB to "push" recordings to PILOS to implement a "pull" mechanism.
- `ImportRecordingsCommand` is now called `withoutOverlapping()`

## Other information

- No tests added because I have no clue how to test this. If you do, let me know and I'll happily implement it. :slightly_smiling_face:

### CODE REVIEW

File: docs/docs/administration/08-advanced/02-recording.md [hunk 1]
```diff
@@ -248,6 +248,16 @@ RECORDING_DESCRIPTION_LIMIT=255
 
 The maximum configurable value is 65,535 characters. As the entire description is displayed in the overview of the recordings, such a high limit is not recommended.
 
+### Recording import hook
+
+If you are unable to push recordings to your PILOS server as described above, you can call let PILOS call a hook script before importing recordings.
```
Comment (samuelwei) [inline docs/docs/administration/08-advanced/02-recording.md]: Wording 'you can call let PILOS call'

**Review by samuelwei — CHANGES_REQUESTED** [review]
I added tests and updated the changelog

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Added
+
+-   Configurable hook script for recording synchronization ([#1484], [#1604])
+
```
Comment (samuelwei) [inline CHANGELOG.md:13]: Add the `withoutOverlapping` change to the `### Changed` section and explain what it does

File: docs/docs/administration/08-advanced/02-recording.md (hunk 1 above)
Comment (pizkaz) [reply docs/docs/administration/08-advanced/02-recording.md]: Fixed.

File: CHANGELOG.md (hunk 2 above)
Comment (pizkaz) [reply CHANGELOG.md:13]: Done.

**Review by samuelwei — APPROVED** [review]

**Pull request merged**